presto-docs/src/main/sphinx/security/internal-communication.rst

```diff
@@ -0,0 +1,80 @@
+=============================
+Secure Internal Communication
+=============================
+
+The Presto cluster can be configured to use secured communication. Communication
+between Presto nodes can be secured with encryption, authentication and authorization.
+
+.. Note: At this step only encryption is supported
+
+Encrypted Internal Communication Configuration
+----------------------------------------------
+
+Encryption is configured in `config.properties` file.
+Encryption properties are identical for all of the presto node roles (coordinator, worker)
+and must be set on every node in the cluster.
+
+To enable encryption for the Presto internal communication follow the next steps:
+
+1. Disable HTTP endpoint.
+
+.. code-block:: none
+
+    http-server.http.enabled=false
+
+.. warning::
+
+    Leaving HTTP endpoint enabled leads to a security hole.
+
+2. Specify node's domains. Every node must have it's own domain configured.
+Domains must be in the same parent domain to be able to easy generate custom SSL
+certificates. e.g.: `coordinator.example.com`, `worker1.example.com`, `worker2.example.com`.
+
+.. code-block:: none
+
+    node.external-address=<node-domain>
+
+3. Generate Java Keystore File.
+
+.. code-block:: none
+
+    keytool -genkeypair -alias example.com -keyalg RSA -keystore keystore.jks
+    Enter keystore password:
+    Re-enter new password:
+    What is your first and last name?
+      [Unknown]:  *.example.com
```

Comment:
Is this literally '*', or should the * be replaced by the hostname you're generating a certificate for?